CRM/Contribute/BAO/Product.php

```diff
@@ -136,16 +136,14 @@ public static function del($productID) {
     $premiumsProduct = new CRM_Contribute_DAO_PremiumsProduct();
     $premiumsProduct->product_id = $productID;
     if ($premiumsProduct->find(TRUE)) {
-      $session = CRM_Core_Session::singleton();
-      $message .= ts('This Premium is being linked to <a href=\'%1\'>Online Contribution page</a>. Please remove it in order to delete this Premium.', array(1 => CRM_Utils_System::url('civicrm/admin/contribute', 'reset=1')), ts('Deletion Error'), 'error');
-      CRM_Core_Session::setStatus($message);
-      return CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/admin/contribute/managePremiums', 'reset=1&action=browse'));
+      throw new CRM_Core_Exception('Cannot delete a Premium that is linked to a Contribution page');
+    }
+    else {
```

Comment:
We don't need this else because an exception will have been thrown. Without this the change looks less scary & I agree that the change to throwing an exception does make sense in the same change that adds a try-catch!